src/User/TeamService.php

```diff
@@ -31,8 +46,67 @@ public function countTeams(): int
         return $this->cache['count'];
     }
 
+    public function createNewTeam(string $name): Team
+    {
+        $team = new Team($name);
+        $this->dispatcher->dispatch(new TeamCreateEvent($team));
+
+        return $team;
+    }
+
+    public function saveTeam(Team $team): Team
+    {
+        if(null === $team->getId()) {
+            // invalidate cache only on new teams
+            return $this->saveNewTeam($team);
+        }
+
+        return $this->updateTeam($team);
+    }
+
+    private function saveNewTeam(Team $team): Team
+    {
+        if (null !== $team->getId()) {
+            throw new InvalidArgumentException('Cannot create team, already persisted');
+        }
+
+        $this->validateTeam($team, ['TeamCreate']);
```

Comment:
Sorry, missed one more... I guess you copied that from User, which is a special case with extra validation groups. Teams donÄt have them:
```suggestion
        $this->validateTeam($team);
```